bin/node-pg-migrate.ts

```diff
@@ -490,9 +490,12 @@ if (action === 'create') {
   if (!DB_CONNECTION) {
     const cp = new ConnectionParameters();
 
-    if (!cp.host && !cp.port && !cp.database) {
+    if (
+      !process.env[argv[databaseUrlVarArg]] &&
+      (!process.env.PGHOST || !cp.user || !cp.database)
```

Comment:
**question (if-minor):** you removed the check for `!cp.host && !cp.port`, but added `!cp.user` and used `||` (or). Can you explain a bit why you did it like that?

Reply:
`cp.host` and `cp.port` are unlikely to be `undefined` since we read `pg` defaults, which set `host` to `'localhost'` and `port` to `'5432'` if missing. 
When running `pg` locally without any env vars, `node-pg-migrate` would run but give an error due to an `undefined` password string/function. Adding a check for the pw alone isn't enough because it would still cause a connection error if `pg` isn't running locally (also when pw is set but others vars aren't). 
so, if there's no `dburl`, I check if any required variable or `process.env.PGHOST` is missing and stop it